frontend/src/components/history/questionHistory.js

```diff
@@ -0,0 +1,152 @@
+    }
+    if (username) {
+      getHistory();
+    }
+  }, [navigate]);
+
+  return (
+    <div>
+      <AttemptInfo open={open} handleClose={handleClose} question={targetQuestion} attempt={targetAttempt}/>
+
+    <div className="history-wrapper">
+      <p className="history-title">Question History</p>
+
+      <TableContainer>
+        <Table sx={{ minWidth: 400 }} size="small" aria-label="simple table">
+
+          {/* Insert table headers */}
+          <TableHead component={Paper} className="history-table-header">
+            <TableRow key="header">
+              <TableCell style={{ fontWeight: 'bold' }}>
+                Question Title
+              </TableCell>
+              <TableCell align="center" style={{ fontWeight: 'bold' }}>
+                Difficulty
+              </TableCell>
+              <TableCell align="center" style={{ fontWeight: 'bold' }}>
+                Language
+              </TableCell>
+              <TableCell align="center" style={{ fontWeight: 'bold' }}>
+                Date of Attempt
+              </TableCell>
+              <TableCell align="center" style={{ fontWeight: 'bold' }}>
+                Collaborated With
+              </TableCell>
+            </TableRow>
+          </TableHead>
+
+          {/* Insert table body content */}
+          <TableBody>
+            {history.length === 0 ? (
+              <TableRow>
+                <TableCell colSpan={5} align="center">
+                  You don't have any collaboration record.
+                </TableCell>
+              </TableRow>
+            ) : (
+              history.map((question) => (
+                <TableRow
+                  key={question.id}
+                  sx={{
+                    '&:last-child td, &:last-child th': { border: 0 },
+                  }}
+                  className="history-table-row"
+                  onClick={() => handleClickOpen(question.id)}
+                >
+                  {/* Add table cells */}
+                  <TableCell component="th" scope="row">
+                    {question.title}
+                  </TableCell>
+                  <TableCell align="center">{question.difficulty}</TableCell>
+                  <TableCell align="center">{question.language}</TableCell>
+                  <TableCell align="center">{question.date}</TableCell>
+                  <TableCell align="center">{question.collaborated}</TableCell>
+                </TableRow>
+              ))
+            )}
+          </TableBody>
+        </Table>
+      </TableContainer>
+
+      {/** Pagination */}
+      <div style={{ display: 'flex', justifyContent: 'center', margin: "10px" }}>
+        <Stack spacing={2}>
+          <Pagination
+            sx={{
+              '& .MuiPaginationItem-root': {
+                '&.Mui-selected': {
+                  background: '#F24E1E',
+                  color: '#ffffff',
+                },
+            },
+            }}
+            count={1}
+            page={1}
```

Comment:
you need to set up pagination here, these values are hardcoded and will always only have one page